apps/gitness/src/pages-v2/repo/repo-summary.tsx

```diff
@@ -0,0 +1,225 @@
+    return decodeGitContent(readmeContent?.content?.data)
+  }, [readmeContent])
+
+  const { data: { body: repoDetails } = {} } = useGetContentQuery({
+    path: '',
+    repo_ref: repoRef,
+    queryParams: { include_commit: true, git_ref: normalizeGitRef(selectedBranch) }
+  })
+
+  const { mutate: createToken } = useCreateTokenMutation(
+    { body: {} },
+    {
+      onSuccess: ({ body: newToken }) => {
+        const tokenData = {
+          identifier: newToken.token?.identifier ?? 'Unknown',
+          lifetime: newToken.token?.expires_at
+            ? new Date(newToken.token.expires_at).toLocaleDateString()
+            : 'No Expiration',
+          token: newToken.access_token ?? 'Token not available'
+        }
+
+        setCreatedTokenData(tokenData)
+        setSuccessTokenDialog(true)
+      }
+    }
+  )
+
+  const handleCreateToken = () => {
+    const body = {
+      identifier: `code_token_${generateAlphaNumericHash(5)}`
+    }
+    createToken({ body })
+  }
+
+  const repoEntryPathToFileTypeMap = useMemo((): Map<string, OpenapiGetContentOutput['type']> => {
+    if (repoDetails?.content?.entries?.length === 0) return new Map()
+    const nonEmtpyPathEntries = repoDetails?.content?.entries?.filter(entry => !!entry.path) || []
+    return new Map(nonEmtpyPathEntries.map((entry: OpenapiContentInfo) => [entry?.path ? entry.path : '', entry.type]))
+  }, [repoDetails?.content?.entries])
+
+  const getSummaryItemType = (type: OpenapiGetContentOutput['type']): SummaryItemType => {
+    if (type === 'dir') {
+      return SummaryItemType.Folder
+    }
+    return SummaryItemType.File
+  }
+
+  useEffect(() => {
+    setLoading(true)
+    if (repoEntryPathToFileTypeMap.size > 0) {
+      pathDetails({
+        queryParams: { git_ref: normalizeGitRef(selectedBranch) },
+        body: { paths: Array.from(repoEntryPathToFileTypeMap.keys()) },
+        repo_ref: repoRef
+      })
+        .then(({ body: response }: { body: RepoPathsDetailsOutput }) => {
+          if (response?.details && response.details.length > 0) {
+            setFiles(
+              response.details.map(
+                (item: GitPathDetails) =>
+                  ({
+                    id: item?.path || '',
+                    type: item?.path
+                      ? getSummaryItemType(repoEntryPathToFileTypeMap.get(item.path))
+                      : SummaryItemType.File,
+                    name: item?.path || '',
+                    lastCommitMessage: item?.last_commit?.message || '',
+                    timestamp: item?.last_commit?.author?.when ? timeAgoFromISOTime(item.last_commit.author.when) : '',
+                    user: { name: item?.last_commit?.author?.identity?.name },
+                    sha: item?.last_commit?.sha && getTrimmedSha(item.last_commit.sha),
+                    path: `/spaces/${spaceId}/repos/${repoId}/code/${gitRef || selectedBranch}/~/${item?.path}`
+                  }) as FileProps
```

Comment:
Defaulting `name` to `''` will allow you to remove the `as FileProps` below
```suggestion
                    user: { name: item?.last_commit?.author?.identity?.name || '' },
                    sha: item?.last_commit?.sha && getTrimmedSha(item.last_commit.sha),
                    path: `/spaces/${spaceId}/repos/${repoId}/code/${gitRef || selectedBranch}/~/${item?.path}`
                  })
```